### DISCUSSION TITLE
Rate Limit on the number of Git Clones

### DISCUSSION THREAD

**Discussion by rahuldan** [community/community#44515] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Hi all,

I was building an application that would require a lot of Git Clone calls to different Github repos. Therefore, wanted to know if there is some rate limit on number of such requests, and if so then can I somehow increase it if I buy the paid Pro plan.

Thanks for your help.
Cheers,

**Comment by nonpawite** [comment]
Hi, 

According to [https://docs.github.com/en/get-started/learning-about-github/githubs-products](), they stated that for "Pro" account type, you can have 1000 more Actions minute per month.

**Reply by rahuldan** [reply]
Hi,

Thanks for your response. Is git clone command from a local machine considered as a Github Actions, because in the docs it states that Actions are used to automate stuff on Github.

Cheers,

**Reply by nonpawite** [reply]
Hi,

From what I understand, 'git clone' doesn't count as GitHub Actions. You can take a look at <https://docs.github.com/en/actions/learn-github-actions/understanding-github-actions>. Also I've read more into document and didn't found anything about limiting 'clone command'. Sorry if there's anything wrong.

**Reply by rahuldan** [reply]
Hi,

No, you are correct, I found the same thing. I was just wondering if there is any rate limit on git clones from a particular IP address. Found some conflicting information from a couple of sources, that's why got the confusion.

Thanks,

**Reply by nonpawite** [reply]
Hi,

I think the best way for you is to contact GitHub support <https://support.github.com/>

Cheers!

**Reply by rahuldan** [reply]
Hi,

Thanks. Yes I posted there. But they aren't responding to my ticket :/
Will post the answer here if they get back.

Cheers,

**Comment by rahuldan** [comment, marked as answer]
Received the answer from the Github support team on this:

"Git operations do not consume part of your API rate limit, as there are no API calls made to GitHub.com via the git client.

We don't have any hard limits for clones, though we may delay requests if they come in fast enough to potentially cause overload on one of our servers (this would be determined by the amount of load being placed on our servers at the time of the clones, and would have to exceptionally high to occur). So while Git operations do have dynamic limits, this might result in slower clones, but shouldn't cause any failures.

In terms of API requests: unauthenticated ones are typically limited to ~5000/hr per IP address, 5000/hr per authenticated user, and significantly lower for anonymous requests (typically ~100/hr or less)."

**Reply by friday** [reply]
It's just 60/hr per IP for unauthenticated requests: https://docs.github.com/en/rest/overview/resources-in-the-rest-api?apiVersion=2022-11-28#rate-limits

There is no higher unauthenticated limit. They say otherwise, but [you need to authenticate with your OAuth App id and and secret to get that "unauthenticated" rate](https://docs.github.com/en/rest/overview/resources-in-the-rest-api?apiVersion=2022-11-28#increasing-the-unauthenticated-rate-limit-for-oauth-apps), which is obviously not applicable when you really need to make unauthenticated requests.

**Comment by friday** [comment]
You should probably ignore the API for this if this app is something you plan to go public with and it needs to run unauthenticated requests.

Your options:
1. Cloning as mentioned has no rate limits officially, [unless GitHub determines your usage "gets so high that it starts affecting our ability to provide service to others"](https://github.com/orgs/community/discussions/24841#discussioncomment-3245596).
1. If you need to get information on tags or branches before "cloning" you can use `git ls-remote https://github.com/user/repo.git`.
2. If this info is enough to replace the clone with a flat archive download, then you can get it as zip or tar.gz: `https://github.com/user/repo/archive/[ref].[extension]` (replace extension with zip or tar.gz (only the latter preserves symlinks) and ref with a git ref like `HEAD` for the main branch, or the name of the branch, tag, or a commit hash. This is faster, but won't work for other git hosts like GitLab (they have another URL).
3. Host a central repository that users can download to avoid downloading/cloning multiple git repos.
